nucypher/blockchain/eth/actors.py

```diff
@@ -959,58 +948,65 @@ def initialize_stake(self,
                                           f"for new stake initialization of {amount}")
 
         # Ensure the new stake will not exceed the staking limit
-        if (self.current_stake + amount) > self.economics.maximum_allowed_locked:
+        if (self.locked_tokens(periods=1) + amount) > self.economics.maximum_allowed_locked:
             raise Stake.StakingError(f"Cannot initialize stake - "
                                      f"Maximum stake value exceeded for {self.checksum_address} "
                                      f"with a target value of {amount}.")
 
         # Write to blockchain
-        new_stake = Stake.initialize_stake(staker=self,
+        new_stake = Stake.initialize_stake(staking_agent=self.staking_agent,
+                                           economics=self.economics,
+                                           checksum_address=self.checksum_address,
                                            amount=amount,
                                            lock_periods=lock_periods)
 
+        # Create stake on-chain
+        receipt = self._deposit(amount=new_stake.value.to_nunits(), lock_periods=new_stake.duration)
+
+        # Log and return Stake instance
+        self.log.info(f"{self.checksum_address} Initialized new stake: {amount} tokens for {lock_periods} periods")
+
         # Update staking cache element
         self.refresh_stakes()
 
-        return new_stake
+        return receipt
 
     @only_me
     def prolong_stake(self,
-                      stake_index: int,
+                      stake: Stake,
                       additional_periods: int = None,
                       expiration: maya.MayaDT = None) -> tuple:
```

Comment:
Return type should be TxReceipt or dict.

Reply:
✔️